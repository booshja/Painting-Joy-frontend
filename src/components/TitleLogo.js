// dependencies
import React from "react";
import styled from "styled-components";
// components
import Navbar from "./Navbar";
import { Link } from "react-router-dom";
// assets
import Logo from "../assets/painting-joy-logo-transparent.png";
// breakpoints
import { breakpoints } from "../breakpoints";

const StyledTitleLogo = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 1rem;

    ${breakpoints("margin-top", "", [{ 1024: 0 }])}
`;

const StyledLogo = styled.img`
    width: 250px;
    height: 250px;

    ${breakpoints("width", "px", [{ 414: 300 }, { 768: 350 }])}
    ${breakpoints("height", "px", [{ 414: 300 }, { 768: 350 }])}
`;

const TitleLogo = () => {
    return (
        <StyledTitleLogo>
            <Link to="/">
                <StyledLogo src={Logo} alt="Painting Joy Mural Co. Logo" />
            </Link>
            <Navbar />
        </StyledTitleLogo>
    );
};

export default TitleLogo;
