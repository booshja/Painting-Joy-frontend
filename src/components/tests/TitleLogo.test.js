import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import TitleLogo from "../TitleLogo";
import CartContext from "../../context/CartContext";
import "jest-styled-components";

/** Smoke Test */
it("renders without crashing", () => {
    render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [] }}>
                <TitleLogo />
            </CartContext.Provider>
        </BrowserRouter>
    );
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [] }}>
                <TitleLogo />
            </CartContext.Provider>
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders without crashing", () => {
    const { getByAltText, getByText } = render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [] }}>
                <TitleLogo />
            </CartContext.Provider>
        </BrowserRouter>
    );

    expect(getByAltText("Painting Joy Mural Co. Logo")).toBeInTheDocument();
    expect(getByText("Home")).toBeInTheDocument();
    expect(getByText("Murals")).toBeInTheDocument();
    expect(getByText("Contact")).toBeInTheDocument();
    expect(getByText("Store")).toBeInTheDocument();
    expect(getByText("Cart (0)")).toBeInTheDocument();
});
