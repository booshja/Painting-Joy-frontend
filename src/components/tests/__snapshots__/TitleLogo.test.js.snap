// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c3 {
  font-family: "News Cycle",sans-serif;
  font-weight: 400;
  font-size: 1.4rem;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
}

.c3:hover:not(.active) {
  -webkit-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
}

.c3.active {
  font-weight: 700;
  color: #207070;
}

.c2 {
  display: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 1rem;
  width: 55%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 1rem;
}

.c1 {
  width: 250px;
  height: 250px;
}

@media screen and (min-width:1024px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    margin-top: 0;
  }
}

@media screen and (min-width:414px) {
  .c1 {
    width: 300px;
  }
}

@media screen and (min-width:768px) {
  .c1 {
    width: 350px;
  }
}

@media screen and (min-width:414px) {
  .c1 {
    height: 300px;
  }
}

@media screen and (min-width:768px) {
  .c1 {
    height: 350px;
  }
}

<div
    class="c0"
  >
    <a
      href="/"
    >
      <img
        alt="Painting Joy Mural Co. Logo"
        class="c1"
        src="painting-joy-logo-transparent.png"
      />
    </a>
    <nav
      class="c2"
    >
      <a
        aria-current="page"
        class="c3 active"
        href="/"
      >
        Home
      </a>
      <a
        class="c3"
        href="/murals"
      >
        Murals
      </a>
      <a
        class="c3"
        href="/contact"
      >
        Contact
      </a>
      <a
        class="c3"
        href="/store"
      >
        Store
      </a>
      <a
        class="c3"
        href="/cart"
      >
        Cart (0)
      </a>
    </nav>
  </div>
</DocumentFragment>
`;
